assets/javascripts/kitten/components/tours/highlight-halo/index.js

```diff
@@ -0,0 +1,184 @@
+    0% {
+      transform: scale(1);
+      opacity: 0.2;
+    }
+    50% {
+      transform: scale(.84);
+    }
+    100% {
+      transform: scale(1);
+      opacity: 0.2;
+    }
+`
+const endingAnimationKeyframes = props =>
+  keyframes`
+    from {
+      transform: scale(1);
+      opacity: 0.2;
+    }
+    to {
+      transform: scale(0);
+      opacity: 0;
+    }
+`
+
+const StyledHighlightHalo = styled.div`
+  position: relative;
+
+  height: ${({ haloSize }) => pxToRem(haloSize)};
+  width: ${({ haloSize }) => pxToRem(haloSize)};
+
+  transform: translate(-50%, -50%);
+
+  > div {
+    position: absolute;
+    left: 0;
+    top: 0;
+
+    opacity: 0;
+    border-radius: 50%;
+    background-color: ${({ haloColor }) => haloColor};
+
+    transform: scale(0);
+
+    ${({ highlightHaloAnimation }) => highlightHaloAnimation};
+  }
+
+  > div:nth-of-type(1) {
+    width: 100%;
+    height: 100%;
+
+    animation-delay: ${({ animationDelay }) => animationDelay + 0.2}s,
+      ${({ animationDelay }) => animationDelay + 0.2 + 0.5}s,
+      ${({ animationDelay, getAnimationDelay }) =>
+        animationDelay + 0.2 + 0.5 + getAnimationDelay}s;
+  }
+  > div:nth-of-type(2) {
+    top: 16.66%;
+    left: 16.66%;
+
+    width: 66.66%;
+    height: 66.66%;
+
+    animation-delay: ${({ animationDelay }) => animationDelay + 0.1}s,
+      ${({ animationDelay }) => animationDelay + 0.1 + 0.5}s,
+      ${({ animationDelay, getAnimationDelay }) =>
+        animationDelay + 0.1 + 0.5 + getAnimationDelay}s;
+  }
+  > div:nth-of-type(3) {
+    top: 33.33%;
+    left: 33.33%;
+
+    width: 33.33%;
+    height: 33.33%;
+
+    animation-delay: ${({ animationDelay }) => animationDelay}s,
+      ${({ animationDelay }) => animationDelay + 0.5}s,
+      ${({ animationDelay, getAnimationDelay }) =>
+        animationDelay + 0.5 + getAnimationDelay}s;
+  }
+`
+
+export class HighlightHalo extends Component {
+  highlightHaloAnimation = () => {
+    var animationEasing = 'ease-in-out'
+
+    if (
+      typeof this.props.animationCycles === 'number' &&
+      this.props.animationCycles >= 0
+    ) {
+      return css`
+        animation-name: ${bubbleAnimationKeyframes},
+          ${breathingAnimationKeyframes}, ${endingAnimationKeyframes};
+        animation-duration: 0.5s, ${this.props.animationCycleDuration}s, 0.1s;
+        animation-timing-function: ${animationEasing}, ${animationEasing},
+          ${animationEasing};
+        animation-iteration-count: 1, ${this.props.animationCycles}, 1;
+      `
+    }
+
+    return css`
+      animation-name: ${bubbleAnimationKeyframes},
+        ${breathingAnimationKeyframes};
+      animation-duration: 0.5s, ${this.props.animationCycleDuration}s;
+      animation-timing-function: ${animationEasing}, ${animationEasing};
+      animation-iteration-count: 1, infinite;
+    `
+  }
+
+  getAnimationDelay = () => {
+    if (
+      typeof this.props.animationCycles == 'number' &&
+      this.props.animationCycles >= 0
```

Comment:
Y'a moyen de raccourcir la condition en faisant 
```js
const animationCycles = parseInt(this.props.animationCycles, 10)

if (animationCycles >= 0) {
  …
}
```
Dans notre cas si `this.props.animationCycles` est une `string`, ça renverra `NaN` qui est de type `number` et non `>=0` :).

Reply:
Ah oui en effet ! j'avais commencé par faire un test sur le `typeof`, puis j'ai ajouté le test `>=` sans penser qu'il pourrait aussi remplacer le `typeof`